### PR TITLE
Fixed mfa resolver always using default auth instance to finalize

### LINKED ISSUE #12265: Multifactor resolver always uses default instance
### Description

In FIRMultFactorResolver defining an instance that is not default does not allow one to finish mfa sign in because it always finalizes with the default instance.

I expected the mfa to finalize, but what actually happened was I got a "user does not belong to this project" error.

### Reproducing the issue

Initialize the default instance of FIRAuth with the info.plist to be some firebase project.

Implement a sign in flow with a different instance and catch the multifactor resolver from it's authentication error.

When you try to finalize the sign in with, it tries to use the wrong instance, which can be verified with a breakpoint or by using
a proxy and observing the wrong api key being used.

### Firebase SDK Version

10.9

### Xcode Version

14.3

### Installation Method

CocoaPods

### Firebase Product(s)

Authentication, Messaging

### Targeted Platforms

iOS

### Relevant Log Output

_No response_

### If using Swift Package Manager, the project's Package.resolved

_No response_

### If using CocoaPods, the project's Podfile.lock

_No response_

### PR DESCRIPTION
**Pull request by Somebodybro**
A pull request to fix #12265

### CODE REVIEW

File: FirebaseAuth/Sources/MultiFactor/FIRMultiFactorResolver.m [hunk 1]
```diff
@@ -85,7 +85,7 @@ - (void)resolveSignInWithAssertion:(nonnull FIRMultiFactorAssertion *)assertion
                                completion(nil, error);
                              }
                            } else {
-                             [FIRAuth.auth
+                             [self->auth
```
Comment (paulb777) [inline FirebaseAuth/Sources/MultiFactor/FIRMultiFactorResolver.m:88]: Should be self.auth

Comment (Somebodybro) [reply FirebaseAuth/Sources/MultiFactor/FIRMultiFactorResolver.m]: fixed wrong syntax.

**Review by paulb777 — APPROVED** [review]

**Comment by renkelvin** [comment]
Thanks @Somebodybro for the fix!

**Review by renkelvin — APPROVED** [review]

**Review by titarenko88 — COMMENTED** [review]
Sam

**Pull request merged**